frontend/javascripts/oxalis/model/volumetracing/volume_annotation_sampling.js

```diff
@@ -0,0 +1,324 @@
+          firstKernelDim < kernelSize[dimensionIndices[0]] && !foundVoxel;
+          firstKernelDim++
+        ) {
+          for (
+            let secondKernelDim = 0;
+            secondKernelDim < kernelSize[dimensionIndices[1]] && !foundVoxel;
+            secondKernelDim++
+          ) {
+            const firstDim = firstVoxelDim + firstKernelDim;
+            const secondDim = secondVoxelDim + secondKernelDim;
+            if (voxelMap[firstDim * constants.BUCKET_WIDTH + secondDim] === 1) {
+              const firstDimInGoalBucket =
+                Math.floor(firstDim * scaleToGoal[dimensionIndices[0]]) +
+                voxelOffset[dimensionIndices[0]];
+              const secondDimInGoalBucket =
+                Math.floor(secondDim * scaleToGoal[dimensionIndices[1]]) +
+                voxelOffset[dimensionIndices[1]];
+              goalVoxelMap[
+                firstDimInGoalBucket * constants.BUCKET_WIDTH + secondDimInGoalBucket
+              ] = 1;
+              foundVoxel = true;
+            }
+          }
+        }
+      }
+    }
+    labeledVoxelMapInTargetResolution.set(goalBucket.zoomedAddress, goalVoxelMap);
+  }
+  return labeledVoxelMapInTargetResolution;
+}
+
+export default function sampleVoxelMapToResolution(
+  labeledVoxelMap: LabeledVoxelsMap,
+  dataCube: DataCube,
+  sourceResolution: Vector3,
+  sourceZoomStep: number,
+  targetResolution: Vector3,
+  targetZoomStep: number,
+  dimensionIndices: DimensionMap,
+  thirdDimensionVoxelValue: number,
+): LabeledVoxelsMap {
+  if (sourceZoomStep < targetZoomStep) {
+    return downsampleVoxelMap(
+      labeledVoxelMap,
+      dataCube,
+      sourceResolution,
+      sourceZoomStep,
+      targetResolution,
+      targetZoomStep,
+      dimensionIndices,
+    );
+  } else if (targetZoomStep < sourceZoomStep) {
+    return upsampleVoxelMap(
+      labeledVoxelMap,
+      dataCube,
+      sourceResolution,
+      sourceZoomStep,
+      targetResolution,
+      targetZoomStep,
+      dimensionIndices,
+      thirdDimensionVoxelValue,
+    );
+  } else {
+    return labeledVoxelMap;
+  }
+}
+
+export function applyVoxelMap(
+  labeledVoxelMap: LabeledVoxelsMap,
+  dataCube: DataCube,
+  cellId: number,
+  get3DAddress: Vector2 => Vector3,
+  numberOfSlicesToApply: number,
+  thirdDimensionIndex: 0 | 1 | 2,
+  shouldOverwrite: boolean = true,
+) {
+  for (const [labeledBucketZoomedAddress, voxelMap] of labeledVoxelMap) {
+    let bucket = dataCube.getOrCreateBucket(labeledBucketZoomedAddress);
+    if (bucket.type === "null") {
+      continue;
+    }
+    bucket.markAndAddBucketForUndo();
+    let data = bucket.getOrCreateData();
+    const thirdDimensionValueInBucket = get3DAddress([0, 0])[2];
+    for (let sliceCount = 0; sliceCount < numberOfSlicesToApply; sliceCount++) {
+      if (thirdDimensionValueInBucket + sliceCount === constants.BUCKET_WIDTH) {
+        // The current slice is in the next bucket in the third direction.
```

Comment:
While reading this code, I'm wondering why it happens that the slice is in the next bucket in the third direction, but this doesn't happen in the x and y direction. Is this taken care of when creating the voxel map? I don't think I completely grasped the voxel map concept in conjunction with the `numberOfSlicesToApply` yet.